### PR TITLE
feat: add islandviewer example

### PR DESCRIPTION
**Pull request by ThHarbig**
Toward #871 

## Change List
 - Added Islandviewer example

## Checklist
 - [ ] Ensure the PR works with all demos on the online editor
 - [ ] Unit tests added or updated
 - [x] Examples added or updated
 - [ ] [Documentation](https://github.com/gosling-lang/gosling-website) updated (e.g., added API functions)
 - [x] Screenshots for visual changes (e.g., new encoding support or UI change on Editor)

### CODE REVIEW

File: editor/example/spec/islandviewer.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
+import type { GoslingSpec } from '@gosling.schema';
+
+const circularRadius = 250;
+const centerRadius = 0.5;
+
+const linearHeight = 120;
+const linearSize = linearHeight / 6;
+const spec: GoslingSpec = {
+    title: 'IslandViewer 4 (Bertelli et al. 2017)',
+    subtitle: 'Salmonella enterica subsp. enterica serovar Typhi Ty2, complete genome.',
```
Comment (sehilyi) [inline editor/example/spec/islandviewer.ts:10]: I think it is helpful to have the reference link here.

```suggestion
    subtitle: 'Salmonella enterica subsp. enterica serovar Typhi Ty2, complete genome.',
      "description": "Reimplementation of https://www.pathogenomics.sfu.ca/islandviewer/accession/NC_004631.1/",
```

File: editor/example/spec/islandviewer.ts [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    assembly: [['NC_004631.1', 4791961]],
+    spacing: 50,
+    views: [
+        {
+            layout: 'circular',
+            static: true,
+            alignment: 'overlay',
+            tracks: [
+                {
+                    data: {
```
Comment (sehilyi) [inline editor/example/spec/islandviewer.ts:23]: To add the outline of the circular track to make it closer to the original demo.

```suggestion
                    "style": { "outlineWidth": 1, "outline": "black"},
                    data: {
```

File: editor/example/spec/islandviewer.ts [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+const spec: GoslingSpec = {
+    title: 'IslandViewer 4 (Bertelli et al. 2017)',
+    subtitle: 'Salmonella enterica subsp. enterica serovar Typhi Ty2, complete genome.',
+    assembly: [['NC_004631.1', 4791961]],
+    spacing: 50,
+    views: [
+        {
+            layout: 'circular',
+            static: true,
+            alignment: 'overlay',
```
Comment (sehilyi) [inline editor/example/spec/islandviewer.ts:18]: To close the gap on the top of the circular visualization

```suggestion
            alignment: 'overlay',
            "spacing": 0.1,
```

**Review by sehilyi — COMMENTED** [review]
Looks great! I just made some minor changes to make it a bit closer to the original demo. 

One thing that we can additionally support in Gosling is support drawing the vertical grid lines, which can be implemented in [grid.ts](https://github.com/gosling-lang/gosling.js/blob/1cda6416a2db2c68c1ec9cf55c85c492a4d46437/src/core/mark/grid.ts#L9-L12).

### Screenshot after change
Screenshot 2023-05-05 at 15 00 48

File: editor/example/spec/islandviewer.ts (hunk 1 above)
Comment (ThHarbig) [reply editor/example/spec/islandviewer.ts:10]: When I apply this change I cannot see the description anywhere, where is it supposed to show up?

Comment (sehilyi) [reply editor/example/spec/islandviewer.ts:10]: It is just a note to the spec and is not supposed to show up somewhere.

**Review by sehilyi — APPROVED** [review]

**Pull request merged**